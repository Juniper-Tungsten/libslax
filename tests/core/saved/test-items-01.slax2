version 1.1;

output-method {
    indent "yes";
}

match /top {
    <out> {
        ns xxx = "http://xml.libslax.org/slax";
        
        var $x = 5;
        
        for-each (1 ... 10) {
            <first> .;
        }
        
        for $i ($x ... count) {
            <second count=count> $i;
        }
        
        for $i (1 - size ... 10 div 3) {
            <third> $i;
        }
        /* Simple test case */
        
        for $i (item) {
            <li item=class> $i;
        }
        /* Test that 'sort' gets relocated */
        
        for $i (item) {
            sort . {
                order "ascending";
            }
            <ul item=class> $i;
        }
        /* Test that this 'sort' is not relocated */
        
        for-each (item) {
            sort . {
                order "ascending";
            }
            <ol item=../class> .;
        }
    }
}
