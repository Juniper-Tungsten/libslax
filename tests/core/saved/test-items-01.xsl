<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://code.google.com/p/libslax/slax" version="1.0" extension-element-prefixes="slax">
  <xsl:output indent="yes"/>
  <xsl:template match="/top">
    <out xmlns:xxx="http://code.google.com/p/libslax/slax">
      <xsl:variable name="x" select="5"/>
      <xsl:for-each xmlns:slax="http://code.google.com/p/libslax/slax" select="slax:build-sequence(1, 10)">
        <first>
          <xsl:value-of select="."/>
        </first>
      </xsl:for-each>
      <xsl:variable name="slax-dot-1" select="."/>
      <xsl:for-each xmlns:slax="http://code.google.com/p/libslax/slax" select="slax:build-sequence($x, count)">
        <xsl:variable name="i" select="."/>
        <xsl:for-each select="$slax-dot-1">
          <second count="{count}">
            <xsl:value-of select="$i"/>
          </second>
        </xsl:for-each>
      </xsl:for-each>
      <xsl:variable name="slax-dot-2" select="."/>
      <xsl:for-each xmlns:slax="http://code.google.com/p/libslax/slax" select="slax:build-sequence(1 - size, 10 div 3)">
        <xsl:variable name="i" select="."/>
        <xsl:for-each select="$slax-dot-2">
          <third>
            <xsl:value-of select="$i"/>
          </third>
        </xsl:for-each>
      </xsl:for-each>
      <!-- Simple test case -->
      <xsl:variable name="slax-dot-3" select="."/>
      <xsl:for-each select="item">
        <xsl:variable name="i" select="."/>
        <xsl:for-each select="$slax-dot-3">
          <li item="{class}">
            <xsl:value-of select="$i"/>
          </li>
        </xsl:for-each>
      </xsl:for-each>
      <!-- Test that 'sort' gets relocated -->
      <xsl:variable name="slax-dot-4" select="."/>
      <xsl:for-each select="item">
        <xsl:sort select="." order="ascending"/>
        <xsl:variable name="i" select="."/>
        <xsl:for-each select="$slax-dot-4">
          <ul item="{class}">
            <xsl:value-of select="$i"/>
          </ul>
        </xsl:for-each>
      </xsl:for-each>
      <!-- Test that this 'sort' is not relocated -->
      <xsl:for-each select="item">
        <xsl:sort select="." order="ascending"/>
        <ol item="{../class}">
          <xsl:value-of select="."/>
        </ol>
      </xsl:for-each>
    </out>
  </xsl:template>
</xsl:stylesheet>
