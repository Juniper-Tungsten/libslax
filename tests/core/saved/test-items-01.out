<?xml version="1.0"?>
<out xmlns:xxx="http://code.google.com/p/libslax/slax">
  <first>1</first>
  <first>2</first>
  <first>3</first>
  <first>4</first>
  <first>5</first>
  <first>6</first>
  <first>7</first>
  <first>8</first>
  <first>9</first>
  <first>10</first>
  <second count="12">5</second>
  <second count="12">6</second>
  <second count="12">7</second>
  <second count="12">8</second>
  <second count="12">9</second>
  <second count="12">10</second>
  <second count="12">11</second>
  <second count="12">12</second>
  <third>-3</third>
  <third>-2</third>
  <third>-1</third>
  <third>0</third>
  <third>1</third>
  <third>2</third>
  <third>3</third>
  <li item="counting">one</li>
  <li item="counting">two</li>
  <li item="counting">three</li>
  <li item="counting">four</li>
  <ul item="counting">four</ul>
  <ul item="counting">one</ul>
  <ul item="counting">three</ul>
  <ul item="counting">two</ul>
  <ol item="counting">four</ol>
  <ol item="counting">one</ol>
  <ol item="counting">three</ol>
  <ol item="counting">two</ol>
</out>
